src/plugins/intel_cpu/src/emitters/snippets/aarch64/jit_gemm_copy_b_emitter.cpp

```diff
@@ -75,17 +85,46 @@ void jit_gemm_copy_b_emitter::emit_impl(const std::vector<size_t>& in, const std
     Xbyak_aarch64::XReg x0(0);
     Xbyak_aarch64::XReg x1(1);
     Xbyak_aarch64::XReg x2(2);
-    h->str(Xbyak_aarch64::XReg(in[0]), pre_ptr(h->sp, -get_vec_length()));
-    h->str(Xbyak_aarch64::XReg(out[0]), pre_ptr(h->sp, -get_vec_length()));
-    h->ldr(x2, post_ptr(h->sp, get_vec_length()));
-    h->ldr(x1, post_ptr(h->sp, get_vec_length()));
+    Xbyak_aarch64::XReg aux_reg(3);
+
+    // Prepare memory pointers with offsets
+    std::vector<size_t> mem_ptrs_idxs{in[0], out[0]};
+    const auto& mem_ptrs = utils::transform_idxs_to_regs(mem_ptrs_idxs);
+
+    // Apply memory offsets to input/output pointers
+    // Reserve space on stack for input/output pointers
+    h->sub(h->sp, h->sp, 2 * get_vec_length());
+
+    // Apply offsets and store pointers on stack
+    for (size_t i = 0; i < mem_ptrs.size(); i++) {
+        const auto& ptr_reg = mem_ptrs[i];
+        int32_t stack_offset = i * get_vec_length();
+
+        if (ov::snippets::utils::is_dynamic_value(m_memory_offsets[i])) {
+            // Dynamic offset: read from runtime parameters
+            size_t runtime_offset = GET_OFF(buffer_offsets) + m_buffer_ids[i] * sizeof(size_t);
+            utils::push_ptr_with_runtime_offset_on_stack(h, stack_offset, ptr_reg, aux_reg, runtime_offset);
+        } else {
+            // Static offset: add compile-time constant
+            utils::push_ptr_with_static_offset_on_stack(h, stack_offset, ptr_reg, m_memory_offsets[i]);
+        }
+    }
+
+    // Load back the adjusted pointers for function call
+    h->ldr(x2, Xbyak_aarch64::ptr(h->sp, 1 * get_vec_length()));  // output pointer
+    h->ldr(x1, Xbyak_aarch64::ptr(h->sp, 0 * get_vec_length()));  // input pointer
+
+    // Set up executor pointer as first argument
     const auto& compiled_kernel = get_compiled_kernel_ptr();
     h->mov(x0, compiled_kernel);
 
     Xbyak_aarch64::XReg func_reg(9);
     h->mov(func_reg, get_execute_function_ptr());
     h->blr(func_reg);
 
+    // Restore stack pointer
+    h->add(h->sp, h->sp, 2 * get_vec_length());
```

Comment:
move up to line 116 as this rsp shift is just for init abi_params, should restore after it's work is done. seems like more clear. what do you think?

Reply:
Done